lib.php

```diff
@@ -52,18 +52,33 @@ function observation_get_course_content_items(\core_course\local\entity\content_
 
 /**
  * Adds an Observation instance.
- * @param object $data
+ * @param object $data form data
  * @return int new observation instance id
  */
 function observation_add_instance($data): int {
     global $DB;
     $cmid = $data->coursemodule;
-    // Insert into DB.
+
+    // Ensure if no text was given it is null (instead of an empty string).
+    if ($data->observerins_editor['text'] === "") {
+        $data->observerins_editor['text'] = null;
```

Comment:
these could likely just be unset($data->observerins_editor['text'] etc. They will not be provided and will default to null in the DB

Reply:
I tried using unset(), but found a weird quirk whereby in the function $DB->update_record() if an value in the array is not set, it will essentially ignored when updating the database. In a scenario where there were instructions in the database and the user were to delete the instructions, by using unset the update_record function will ignore the updated instructions when updating the database (leaving the instructions there), rather than the intended behavior of updating the instructions to null.